apollo-router/src/plugins/rhai/mod.rs

```diff
@@ -597,60 +459,19 @@ macro_rules! gen_map_response {
         $borrow.replace(|service| {
             service
                 .map_response(move |response: $base::Response| {
-                    // Let's define a local function to build an error response
-                    // XXX: This isn't ideal. We already have a response, so ideally we'd
-                    // like to append this error into the existing response. However,
-                    // the significantly different treatment of errors in different
-                    // response types makes this extremely painful. This needs to be
-                    // re-visited at some point post GA.
```

Comment:
I'm not sure if this comment is relevant anymore. I remember I wrote it and it meant something, so losing the context might mean we forget to fix something, but ....